samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java

```diff
@@ -271,45 +281,46 @@ public Object convertToJavaObject(Object avroObj, Schema schema) {
     }
   }
 
-  private boolean isSchemaCompatible(Object avroObj, Schema unionSchema) {
-    if (unionSchema.getType() == Schema.Type.NULL && avroObj == null) {
-      return true;
+  private static boolean isSchemaCompatibleWithRelObj(Object relObj, Schema unionSchema) {
+    if (unionSchema.getType() == Schema.Type.NULL) {
```

Comment:
verify/handle unionSchema != null?

Reply:
I believe you are asking whether we should handle null values for unionSchema?

Reply:
I was actually wondering if we should guard against NullPointerException when  unionSchema of Schema is null